frontend/src/components/Chat/ChatSidebar.tsx

```diff
@@ -1,48 +1,181 @@
+  const onDirectChatSelect = useCallback(() => setChatName('Direct Chats'), [setChatName]);
+  const onDirectChatOpen = useCallback(userName => setChatName(`Direct Chat with ${userName}`), [
+    setChatName,
+  ]);
+
+  const townMessageNotifs = useMemo(() => {
+    if (!numberUnviewedTownMessages) {
+      return null;
+    }
+    return (
+      <Tag size='sm' data-testid='town-notifs' textAlign='center' marginLeft='5px' bg='lightblue'>
+        {numberUnviewedTownMessages}
+      </Tag>
+    );
+  }, [numberUnviewedTownMessages]);
+
+  const proximityMessageNotifs = useMemo(() => {
+    if (!numberUnviewedProximityMessages) {
+      return null;
+    }
+    return (
+      <Tag
+        data-testid='proximity-notifs'
+        size='sm'
+        textAlign='center'
+        marginLeft='5px'
+        bg='lightblue'>
+        {numberUnviewedProximityMessages}
+      </Tag>
+    );
+  }, [numberUnviewedProximityMessages]);
+
+  const directMessageNotifs = useMemo(() => {
+    if (!numberUnviewedDirectMessages) {
+      return null;
+    }
+    return (
+      <Tag size='sm' data-testid='direct-notifs' textAlign='center' marginLeft='5px' bg='lightblue'>
+        {numberUnviewedDirectMessages}
+      </Tag>
+    );
+  }, [numberUnviewedDirectMessages]);
 
-  const handleReturnToSelect = () => {
+  const handleReturnToSelect = useCallback(() => {
     if (isViewingDirectChatContainer) {
       setIsViewingDirectChatContainer(false);
+      onDirectChatSelect();
     }
-  };
+  }, [isViewingDirectChatContainer, setIsViewingDirectChatContainer, onDirectChatSelect]);
 
   const directMessageTab = isViewingDirectChatContainer ? 'All Direct Chats' : 'Direct Chats';
 
+  const handleToggleExpandCollapse = useCallback(() => {
+    setIsCollapsed(!isCollapsed);
+  }, [isCollapsed, setIsCollapsed]);
+
+  const collapseExpandButton = useMemo(() => {
+    const messageCount =
+      numberUnviewedDirectMessages + numberUnviewedProximityMessages + numberUnviewedTownMessages;
+    const messageCountToRender = messageCount > 100 ? '100+' : messageCount;
+    const messageLabel = messageCount === 1 ? 'Message' : 'Messages';
+    let symbol;
+    if (isCollapsed) {
+      symbol = '‹';
+    } else {
+      symbol = '›';
+    }
+    return (
+      <Button
+        data-testid='collapse-expand-button'
+        onClick={handleToggleExpandCollapse}
+        bg='lightblue'
+        className='collapse-button'>
+        <span
+          style={{
+            fontSize: '30px',
+            paddingBottom: '5px',
+            paddingRight: isCollapsed ? '5px' : undefined,
+          }}>
+          {symbol}
+        </span>
+        {isCollapsed ? ` ${messageCountToRender} ${messageLabel}` : null}
+      </Button>
+    );
+  }, [
+    handleToggleExpandCollapse,
+    isCollapsed,
+    numberUnviewedTownMessages,
+    numberUnviewedProximityMessages,
+    numberUnviewedDirectMessages,
+  ]);
+
+  const chatSidebarContent = useMemo(

```

Comment:
memoized the sidebar so that it doesn't re-render from scratch on every expand/collapse